src/web/app/pages-session/session-submission-page/session-submission-page.component.ts

```diff
@@ -546,9 +550,13 @@ export class SessionSubmissionPageComponent implements OnInit, AfterViewInit {
 
     this.questionSubmissionForms.forEach((questionSubmissionFormModel: QuestionSubmissionFormModel) => {
       let isQuestionFullyAnswered: boolean = true;
-
       questionSubmissionFormModel.recipientSubmissionForms
           .forEach((recipientSubmissionFormModel: FeedbackResponseRecipientSubmissionFormModel) => {
+            if (!recipientSubmissionFormModel.isValid) {
+              failToSaveQuestions[questionSubmissionFormModel.questionNumber] =
+                  'Invalid responses provided. Please check question constraints.';
+              return;
```

Comment:
Yes, because the `isValid` check is in the `FeedbackResponseRecipientSubmissionFormModel` per-response, which is embedded in the `QuestionSubmissionFormModel`. This is the natural result of implementing per-response validation instead of per-question